pkg/kubelet/kuberuntime/logs/logs_test.go

```diff
@@ -210,6 +212,138 @@ func TestReadLogs(t *testing.T) {
 	}
 }
 
+type syncBuffer struct {
+	b bytes.Buffer
+	sync.Mutex
+}
+
+func (b *syncBuffer) Read(p []byte) (n int, err error) {
+	b.Lock()
+	defer b.Unlock()
+	return b.b.Read(p)
+}
+func (b *syncBuffer) Write(p []byte) (n int, err error) {
+	b.Lock()
+	defer b.Unlock()
+	return b.b.Write(p)
+}
+func (b *syncBuffer) String() string {
+	b.Lock()
+	defer b.Unlock()
+	return b.b.String()
+}
+
+func TestReadRotatedLog(t *testing.T) {
+	file, err := os.CreateTemp("/tmp", "TestFollowLogs")
```

Comment:
will it make test to watch the whole `/tmp`? 

Also, maybe rely on the default temp folder instead of setting to `/tmp` explicitly

Reply:
Done.